src/front/spirv.rs

```diff
@@ -707,618 +681,36 @@ impl<I: Iterator<Item = u32>> Parser<I> {
-                            return_type: if self.lookup_void_type.contains(&result_type) {
-                                None
-                            } else {
-                                Some(self.lookup_type.lookup(result_type)?.handle)
-                            },
-                            local_variables: Arena::new(),
-                            expressions: self.make_expression_storage(),
-                            body: Vec::new(),
-                        }
-                    };
-                    // read parameters
-                    for i in 0 .. fun.parameter_types.capacity() {
-                        match self.next_inst()? {
-                            Instruction { op: Op::FunctionParameter, wc: 3 } => {
-                                let type_id = self.next()?;
-                                let _id = self.next()?;
-                                //Note: we redo the lookup in order to work around `self` borrowing
-                                if type_id != self.lookup_function_type
-                                    .lookup(fun_type)?
-                                    .parameter_type_ids[i]
-                                {
-                                    return Err(Error::WrongFunctionParameterType(type_id))
-                                }
-                                let ty = self.lookup_type.lookup(type_id)?.handle;
-                                fun.parameter_types.push(ty);
-                            }
-                            Instruction { op, .. } => return Err(Error::InvalidParameter(op)),
-                        }
-                    }
-                    // read body
-                    loop {
-                        let fun_inst = self.next_inst()?;
-                        log::debug!("\t\t{:?}", fun_inst.op);
-                        match fun_inst.op {
-                            Op::Label => {
-                                fun_inst.expect(2)?;
-                                let _id = self.next()?;
-                                self.next_block(&mut fun, &module.types, &module.constants)?;
-                            }
-                            Op::FunctionEnd => {
-                                fun_inst.expect(1)?;
-                                break
-                            }
-                            _ => return Err(Error::UnsupportedInstruction(self.state, fun_inst.op))
-                        }
-                    }
-                    // done
-                    let handle = module.functions.append(fun);
-                    self.lookup_function.insert(fun_id, handle);
-                    self.lookup_expression.clear();
-                    self.lookup_sampled_image.clear();
-                }
-                _ => return Err(Error::UnsupportedInstruction(self.state, inst.op))
-                //TODO
-            }
+                Op::Capability => self.parse_capability(inst),
```

Comment:
is that something Clippy complained about?

Reply:
It complained this function was too complicated, this seemed like the most sensible transformation to fix that.